rasa/nlu/components.py

```diff
@@ -87,13 +87,40 @@ def validate_required_components_from_data(
     pipeline: List["Component"], data: TrainingData
 ):
 
-    response_selector_exists = False
-    for component in pipeline:
-        # check if a response selector is part of NLU pipeline
-        if MESSAGE_RESPONSE_ATTRIBUTE in component.provides:
-            response_selector_exists = True
+    # check if there are regex features
+    if data.regex_features and not any(
+        [component.name == "RegexFeaturizer" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data with Regex's, but "
+            "haven't setup RegexFeaturizer in your config. "
+            "Either add RegexFeaturizer to your config or "
+            "remove the regex training data"
+        )
+
+    if data.lookup_tables and not any(
+        [component.name == "CRFEntityExtractor" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data consisting of lookup tables but "
+            "haven't setup CRFEntityExtractor in your config. "
+            "Either add CRFEntityExtractor to your config or "
+            "remove the lookup table training data"
+        )
+
+    if data.entity_synonyms and not any(
+        [component.name == "EntitySynonymMapper" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data consisting of synonyms but "
+            "haven't setup EntitySynonymMapper in your config. "
+            "Either add EntitySynonymMapper to your config or "
+            "remove the synonym training data"
+        )
 
-    if len(data.response_examples) and not response_selector_exists:
+    if len(data.response_examples) and not any(
+        [MESSAGE_RESPONSE_ATTRIBUTE in component.provides for component in pipeline]
+    ):
         warnings.warn(
             "Training data consists examples for training a response selector but "
             "no response selector component specified inside NLU pipeline"
```

Comment:
This one however should get the "either add component or remove training data" as it will break stuff

Reply:
have updated this messaging

Reply:
@btotharye feel free to just resolve conversations when they are updated instead of commenting on each :)